contracts/bridge/contracts/common/message/InboundLane.sol

```diff
@@ -0,0 +1,301 @@
+            Message memory message = messages[i];
+            MessageKey memory key = message.key;
+            MessagePayload memory message_payload = message.data.payload;
+            if (key.nonce < next) {
+                continue;
+            }
+            // check message nonce is correct and increment nonce for replay protection
+            require(key.nonce == next, "Lane: InvalidNonce");
+            // check message is from the correct source chain position
+            require(key.this_chain_id == bridgedChainPosition, "Lane: InvalidSourceChainId");
+            // check message delivery to the correct chain position
+            require(key.bridged_chain_id == thisChainPosition, "Lane: InvalidTargetChainId");
+            // check message delivery to the correct lane position
+            require(key.lane_id == lanePosition, "Lane: InvalidLaneID");
+            // if there are more unconfirmed messages than we may accept, reject this message
+            require(next - inboundLaneNonce.last_confirmed_nonce <= MAX_UNCONFIRMED_MESSAGES, "Lane: TooManyUnconfirmedMessages");
+
+            // then, dispatch message
+            (bool dispatch_result, bytes memory returndata) = dispatch(message_payload);
+
+            emit MessageDispatched(thisChainPosition, lanePosition, next, dispatch_result, returndata);
+            // TODO: callback `pay_inbound_dispatch_fee_overhead`
+            dispatch_results |= (dispatch_result ? uint256(1) : uint256(0)) << i;
+            end = next;
+            next += 1;
+        }
+        if (end > inboundLaneNonce.last_delivered_nonce) {
+            // update inbound lane nonce storage
+            inboundLaneNonce.last_delivered_nonce = end;
+
+            // now let's update inbound lane storage
+            address pre_relayer = relayers_back();
+            if (pre_relayer == relayer) {
+                UnrewardedRelayer storage r = relayers[relayersRange.back];
+                r.messages.dispatch_results |= dispatch_results << (r.messages.end - r.messages.begin + 1);
+                r.messages.end = end;
+            } else {
+                relayersRange.back += 1;
+                relayers[relayersRange.back] = UnrewardedRelayer(relayer, DeliveredMessages(begin, end, dispatch_results));
+            }
+        }
+    }
+
+    function dispatch(MessagePayload memory payload) internal returns (bool dispatch_result, bytes memory returndata) {
+        bytes memory filterCallData = abi.encodeWithSelector(
+            ICrossChainFilter.crossChainFilter.selector,
+            bridgedChainPosition,
+            payload.sourceAccount,
+            payload.encoded
+        );
+        bool canCall = filter(payload.targetContract, filterCallData);
+        if (canCall) {
+            // Deliver the message to the target
+            (dispatch_result, returndata) = payload.targetContract.call{value: 0, gas: MAX_GAS_PER_MESSAGE}(payload.encoded);
```

Comment:
if the call used a large number of gas. This gas will cause the dispatch revert. It will never be verified, and the messages which nonce bigger will also be verified?